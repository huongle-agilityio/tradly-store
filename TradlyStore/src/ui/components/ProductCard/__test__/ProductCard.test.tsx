import { render, fireEvent, screen } from '@testing-library/react-native';

// Components
import { ProductCard } from '..';

describe('ProductCard Component', () => {
  const mockProps = {
    source: 'image-url',
    title: 'Product Title',
    storeName: 'Store Name',
    storeSource: 'store-logo-url',
    price: 100,
    onPress: jest.fn(),
  };

  it('Should renders correctly with required props', () => {
    const { toJSON } = render(<ProductCard {...mockProps} discount={20} />);

    expect(screen.getByTestId('category-card-image')).toBeTruthy();
    expect(screen.getByText(mockProps.title)).toBeTruthy();
    expect(screen.getByText(mockProps.storeName)).toBeTruthy();
    expect(screen.getByText('$80')).toBeTruthy();
    expect(screen.getByText('$100')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should calls onPress when clicked', () => {
    render(<ProductCard {...mockProps} />);
    fireEvent.press(screen.getByTestId('product-card'));

    expect(mockProps.onPress).toHaveBeenCalled();
  });

  it('Should hides original price if no discount is applied', () => {
    render(<ProductCard {...mockProps} />);

    expect(screen.getByText('$100')).toBeTruthy();
  });
});
