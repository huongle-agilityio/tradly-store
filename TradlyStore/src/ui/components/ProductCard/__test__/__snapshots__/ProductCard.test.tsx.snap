// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component Should renders correctly with required props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#dbdbde",
      "borderRadius": 8,
      "borderWidth": 1,
      "opacity": 1,
      "width": 160,
    }
  }
  testID="product-card"
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(8, 11, 18, 0.3)",
        },
        {
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
        },
      ]
    }
  >
    <Image
      alt="product-Product Title"
      source={
        {
          "uri": "image-url",
        }
      }
      style={
        [
          {
            "elevation": 2,
            "height": 125,
            "position": "relative",
            "width": "100%",
          },
          {
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
        ]
      }
      testID="category-card-image"
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "gap": 14,
        "padding": 12,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "fontSize": 14,
            "lineHeight": 16,
          },
          {
            "fontFamily": "Montserrat-Medium",
            "fontWeight": "500",
          },
          {
            "color": "#4a4a4a",
          },
          {
            "lineHeight": 18,
          },
        ]
      }
    >
      Product Title
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 5,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 6,
          }
        }
      >
        <Image
          alt="store-Store Name-logo"
          source={
            {
              "uri": "store-logo-url",
            }
          }
          style={
            {
              "borderRadius": 9999,
              "height": 20,
              "width": 20,
            }
          }
        />
        <Text
          numberOfLines={1}
          style={
            [
              {
                "fontSize": 14,
                "lineHeight": 16,
              },
              {
                "fontFamily": "Montserrat-Medium",
                "fontWeight": "500",
              },
              {
                "color": "#4f4f4f",
              },
              {
                "lineHeight": 18,
                "width": 40,
              },
            ]
          }
        >
          Store Name
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 6,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 10,
                "lineHeight": 12,
              },
              {
                "fontFamily": "Montserrat-Medium",
                "fontWeight": "500",
              },
              {
                "color": "#4f4f4f",
              },
              {
                "textDecorationLine": "line-through",
              },
            ]
          }
        >
          $
          100
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 14,
                "lineHeight": 16,
              },
              {
                "fontFamily": "Montserrat-Medium",
                "fontWeight": "500",
              },
              {
                "color": "#33907c",
              },
              undefined,
            ]
          }
        >
          $
          80
        </Text>
      </View>
    </View>
  </View>
</View>
`;
